tests/model_inheritance/tests.py

```diff
@@ -72,6 +72,16 @@ def test_model_with_distinct_accessors(self):
             AttributeError, getattr, post, "attached_%(class)s_set"
         )
 
+    def test_model_with_distinct_related_query_name(self):
+        post = Post.objects.create(title="Lorem Ipsum")
+        self.assertQuerysetEqual(Post.objects.filter(attached_comments__is_spam=True), [])
+
+        # The Post model doesn't have an attribute called
+        # 'attached_%(class)ss'.
+        self.assertRaises(
```

Comment:
please use the context manager version: `with self.assertRaises(AttributeError):` (there's another ticket open to change existing usages)